libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/app/MobileSyncUpgradeManager.java

```diff
@@ -66,20 +67,21 @@ public void upgrade() {
      * Upgrades mobile sync data from existing client version to the current version.
      */
     protected synchronized void upgradeSObject() {
-        final String installedVersion = getInstalledSobjectVersion();
-        if (installedVersion.equals(MobileSyncSDKManager.SDK_VERSION)) {
+        final String installedVersionStr = getInstalledSobjectVersion();
+        if (installedVersionStr.equals(MobileSyncSDKManager.SDK_VERSION)) {
             return;
         }
 
         // Update shared preference file to reflect the latest version.
         writeCurVersion(MOBILE_SYNC_KEY, MobileSyncSDKManager.SDK_VERSION);
 
-        // If the installed version < v8.2.0, we need to delete the old layout SmartStore file.
         try {
```

Comment:
What's the point of that try-catch? We are not doing anything with the calculated installedVersion.

Reply:
It was there before, but I could remove it if that's preferred.

Reply:
I make `SdkVersion` also throw `IllegalArgumentException` if the format of the version string is not recognized, so I think the pattern of doing the try-catch to parse the version is good.  But in these files where the try-catch is empty, it does not serve any purpose to keep them.  They were kept in there in previous versions, so I just left them this time too.

Reply:
The `MobileSyncUpgradeManager`'s `upgrade` calls its super class `SmartStoreUpgradeManager`'s `upgrade` which calls its super class `SalesforceSDKUpgradeManager`'s `upgrade which does the same parsing in a try-catch. So I would say let's drop the empty try-catch in the subclasses that don't need to do anything with the calculated version.
We could even refactor the code a bit to do that parsing once, maybe make that installed version available in a member?